darts/timeseries.py

```diff
@@ -705,6 +730,106 @@ def update(self,
             new_series = new_series.astype(self._df.dtypes)
         return TimeSeries(new_series, self.freq())
 
+    def stack(self, other: 'TimeSeries') -> 'TimeSeries':
+        """
+        Stacks another univariate or multivariate TimeSeries with the same index on top of
+        the current one and returns the newly formed multivariate TimeSeries.
+
+        Parameters
+        ----------
+        other
+            A TimeSeries instance with the same index as the current one.
+
+        Returns
+        -------
+        TimeSeries
+            A new multivariate TimeSeries instance.
+        """
+        raise_if_not((self.time_index() == other.time_index()).all(), 'The indices of the two TimeSeries instances '
+                     'must be equal', logger)
+
+        new_dataframe = pd.concat([self.pd_dataframe(), other.pd_dataframe()], axis=1)
+        return TimeSeries(new_dataframe, self.freq())
+
+    @property
+    def width(self) -> int:
+        """
+        Returns
+        -------
+        int
+            The number of components (univariate time series) of the current TimeSeries instance.
+        """
+        return self._df.shape[1]
+
+    def univariate_component(self, index: int) -> 'TimeSeries':
+        """
+        Retrieves one of the components of the current TimeSeries instance
+        and returns it as new univariate TimeSeries instance.
+
+        Parameters
+        ----------
+        index
+            An zero-indexed integer indicating which component to retrieve.
+
+        Returns
+        -------
+        TimeSeries
+            A new univariate TimeSeries instance.
+        """
+
+        raise_if_not(index >= 0 and index < self.width, 'The index must be between 0 and the number of components '
+                     'of the current TimeSeries instance - 1, {}'.format(self.width - 1), logger)
+
+        return TimeSeries.from_series(self.pd_dataframe().iloc[:, index], self.freq())
+
+    def add_datetime_attribute(self, attribute: str, one_hot: bool = False) -> 'TimeSeries':
```

Comment:
I wonder if this shouldn't be in `utils` rather than in `TimeSeries`, wdyt? (And same for `add_holidays`).